tests/test_eval.cpp

```diff
@@ -96,4 +97,15 @@ TEST_SUBMODULE(eval_, m) {
         auto int_class = py::eval("isinstance(42, int)", global);
         return global;
     });
+
+    m.def("test_eval_closure", [global, copy]() {
+        py::dict local = copy(global);
+        py::exec(R"(
+            closure_value = 10
+
+            def func():
+                return closure_value
+            )", local, local);
```

Comment:
Why is `local` being passed twice here? And from reading #1742, shouldn't this be called `global` as it's the globals dict?